### ISSUE TITLE
Introduce general helper function to point elements from warnings

### ISSUE THREAD

**Issue by saschanaz** [w3c/respec#1944]
See https://github.com/w3c/respec/pull/1943#discussion_r240034603 for the context.

I would be happy if the warning/error list can point to the problematic element if possible, rather than only the messages.

**Comment by marcoscaceres** [comment]
Agree... we need something like `authorWarn(msg, optional elem)`. And if elem is passed we `console.warn` and add the "respec-offending-element" CSS class to it.
